hummingbot/connector/exchange/bittrex/bittrex_api_user_stream_data_source.py

```diff
@@ -1,160 +1,98 @@
 import asyncio
-import hashlib
-import hmac
-import logging
-import time
-import uuid
-
-from base64 import b64decode
-from typing import Any, AsyncIterable, Dict, List, Optional
-from zlib import decompress, MAX_WBITS
 
 import signalr_aio
-import ujson
-
-from async_timeout import timeout
 
+from hummingbot.connector.exchange.bittrex import bittrex_constants as CONSTANTS
 from hummingbot.connector.exchange.bittrex.bittrex_auth import BittrexAuth
+from hummingbot.connector.exchange.bittrex.bittrex_utils import _socket_stream, _transform_raw_message
 from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
-from hummingbot.logger import HummingbotLogger
-
-BITTREX_WS_FEED = "https://socket-v3.bittrex.com/signalr"
-MAX_RETRIES = 20
-MESSAGE_TIMEOUT = 30.0
-NaN = float("nan")
 
 
 class BittrexAPIUserStreamDataSource(UserStreamTrackerDataSource):
-
-    MESSAGE_TIMEOUT = 30.0
-    PING_TIMEOUT = 10.0
-
-    _btausds_logger: Optional[HummingbotLogger] = None
-
-    @classmethod
-    def logger(cls) -> HummingbotLogger:
-        if cls._btausds_logger is None:
-            cls._btausds_logger = logging.getLogger(__name__)
-        return cls._btausds_logger
-
-    def __init__(self, bittrex_auth: BittrexAuth, trading_pairs: Optional[List[str]] = []):
-        self._bittrex_auth: BittrexAuth = bittrex_auth
-        self._trading_pairs = trading_pairs
-        self._current_listen_key = None
-        self._listen_for_user_stream_task = None
-        self._last_recv_time: float = 0
-        self._websocket_connection: Optional[signalr_aio.Connection] = None
-        self._hub = None
+    LOCK = asyncio.Lock()
+    INVOCATION_EVENT = None
+    INVOCATION_RESPONSE = None
+
+    def __init__(self,
+                 auth: BittrexAuth,
+                 connector,
+                 api_factory):
         super().__init__()
-
-    @property
-    def hub(self):
-        return self._hub
-
-    @hub.setter
-    def hub(self, value):
-        self._hub = value
+        self._auth = auth
+        self._connector = connector
+        self._api_factory = api_factory
+        self.hub = None
 
     @property
     def last_recv_time(self) -> float:
-        return self._last_recv_time
-
-    async def _socket_user_stream(self, conn: signalr_aio.Connection) -> AsyncIterable[str]:
+        """
+        Overriding method for Bittrex Signalr
+        """
+        if self._ws_assistant:
+            return self._auth.time_provider.time()
+        return 0
+
+    async def _connected_websocket_assistant(self):
+        websocket_connection = signalr_aio.Connection(CONSTANTS.BITTREX_WS_URL, session=None)
+        self.hub = websocket_connection.register_hub("c3")
+        websocket_connection.received += self.handler
+        websocket_connection.error += self.handler
```

Comment:
The same comment as in the order book data source: it does not seem correct to use the same handler for normal messages and errors.

Reply:
Reworked the implementation of signalr with inspiration from the CoinDCX implementation.